tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,494 @@
+import logging
+import pytest
+
+import homeassistant.components.media_player as mp
+from homeassistant.components.media_player import (
+    ATTR_MEDIA_CONTENT_ID, ATTR_MEDIA_CONTENT_TYPE, ATTR_MEDIA_ENQUEUE, DOMAIN,
+    SERVICE_PLAY_MEDIA)
+
+from homeassistant.components.media_player.directv import (
+    ATTR_MEDIA_CURRENTLY_RECORDING, ATTR_MEDIA_RATING, ATTR_MEDIA_RECORDED,
+    ATTR_MEDIA_START_TIME, DATA_DIRECTV, DEFAULT_DEVICE, DEFAULT_PORT,
+    setup_platform)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_DEVICE, CONF_HOST, CONF_NAME, CONF_PORT,
+    SERVICE_MEDIA_NEXT_TRACK, SERVICE_MEDIA_PAUSE, SERVICE_MEDIA_PLAY,
+    SERVICE_MEDIA_PREVIOUS_TRACK, SERVICE_MEDIA_STOP, SERVICE_TURN_OFF,
+    SERVICE_TURN_ON, STATE_OFF, STATE_PAUSED, STATE_PLAYING)
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import MockDependency
+
+CLIENT_ENTITY_ID = 'media_player.client_dvr'
+MAIN_ENTITY_ID = 'media_player.main_dvr'
+IP_ADDRESS = '127.0.0.1'
+
+DISCOVERY_INFO = {
+    'host': IP_ADDRESS,
+    'serial': 1234
+}
+
+LIVE = {
+    "callsign": "CNNHD",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": False,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield"
+}
+
+LOCATIONS = [
+    {
+        'locationName': 'Main DVR',
+        'clientAddr': DEFAULT_DEVICE
+    }
+]
+
+RECORDING = {
+    "callsign": "CNNHD",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": True,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield",
+    'uniqueId': '12345',
+    'episodeTitle': 'CNN Recorded'
+}
+
+WORKING_CONFIG = {
+    'media_player': {
+        'platform': 'directv',
+        CONF_HOST: IP_ADDRESS,
+        CONF_NAME: 'Main DVR',
+        CONF_PORT: DEFAULT_PORT,
+        CONF_DEVICE: DEFAULT_DEVICE
+    }
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def platforms():
+    """Fixture for setting up test platforms."""
+    config = {
+        'media_player': [{
+            'platform': 'directv',
+            'name': 'Main DVR',
+            'host': IP_ADDRESS,
+            'port': DEFAULT_PORT,
+            'device': DEFAULT_DEVICE
+        }, {
+            'platform': 'directv',
+            'name': 'Client DVR',
+            'host': IP_ADDRESS,
+            'port': DEFAULT_PORT,
+            'device': '1'
+        }]
+    }
+
+    async def setup_test_platforms(hass):
```

Comment:
Why make a factory? Instead `yield` from the fixture to keep the patch in place during the test.

Reply:
I used it to be able to return the entity objects. Changed now.